### PR TITLE
python-packages: use findutils from buildPackages in hooks

### PR DESCRIPTION
**Pull request by yorickvP**
This fixes a bunch of errors when cross-compiling.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Any example you can give that I can check out?

**Comment by yorickvP** [comment]
The python namespaces hook is broken without this. Try `pkgsCross.aarch64-multiplatform.python3.pkgs.jaraco_stream`:
```
       > /nix/store/f4hyzz5crxafywc4m1xzzmnn7akynw73-python-namespaces-hook.sh/nix-support/setup-hook: line 35: /nix/store/fvg384csmrynjf8vww44ga65fnxkvf8w-findutils-aarch64-unknown-linux-gnu-4.9.0/bin/find: cannot execute binary file: Exec format error
```

**Comment by mweinelt** [comment]
That would be helpful in the commit message, can you add it?

**Review by mweinelt — APPROVED** [review]

**Comment by Artturin** [comment]
i cant push in to this branch so please do this instead

```
diff --git a/pkgs/development/interpreters/python/hooks/default.nix b/pkgs/development/interpreters/python/hooks/default.nix
index 5d605b240ad..ca445afce3c 100644
--- a/pkgs/development/interpreters/python/hooks/default.nix
+++ b/pkgs/development/interpreters/python/hooks/default.nix
@@ -5,7 +5,6 @@
 , disabledIf
 , isPy3k
 , ensureNewerSourcesForZipFilesHook
-, findutils
 , installShellFiles
 }:
 
@@ -107,7 +106,7 @@ in rec {
       };
     } ./python-imports-check-hook.sh) {};
 
-  pythonNamespacesHook = callPackage ({}:
+  pythonNamespacesHook = callPackage ({ findutils }:
     makeSetupHook {
       name = "python-namespaces-hook.sh";
       substitutions = {
```

**Comment by yorickvP** [comment]
Done, and checked that it still works.

File: pkgs/development/interpreters/python/hooks/default.nix [hunk 1]
```diff
@@ -107,7 +106,7 @@ in rec {
       };
     } ./python-imports-check-hook.sh) {};
 
-  pythonNamespacesHook = callPackage ({}:
+  pythonNamespacesHook = callPackage ({ findutils }:
     makeSetupHook {
       name = "python-namespaces-hook.sh";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/python/hooks/default.nix:111]:
```suggestion
      name = "python-namespaces-hook.sh";
      deps = [ findutils ];
```
This is how all the other hooks are doing it.

**Comment by mweinelt** [comment]
Touching widely used python hooks will need to go through staging.

File: pkgs/development/interpreters/python/hooks/default.nix (hunk 1 above)
Comment (Artturin) [reply pkgs/development/interpreters/python/hooks/default.nix:111]: not necessary, please undo this change. findutils is already substituted below

File: pkgs/development/interpreters/python/hooks/default.nix [hunk 2]
```diff
@@ -107,9 +106,10 @@ in rec {
...
+      deps = [ findutils ];
```
Comment (Artturin) [inline pkgs/development/interpreters/python/hooks/default.nix:112]: not necessary, please undo this change. findutils is already substituted below. without this change this pr doesn't rebuild anything so can go to master

Comment (FRidh) [reply pkgs/development/interpreters/python/hooks/default.nix:112]: See also https://github.com/NixOS/nixpkgs/pull/192624
